### ISSUE TITLE
Allow sparse tensors to be densified by dimension to give hybrid sparse tensors

### ISSUE THREAD

**Issue by ej159** [pytorch/pytorch#107293]
### 🚀 The feature, motivation and pitch

I work with tensors that are very sparse in one dimension (the row) but also quite sparse in another dimension (column). I want to be able to go between being fully sparse and sparse-dense hybrid tensors easily. 

Would it be possible to add a "dim" keywords to to_dense and to_sparse to specify which dimension is being made dense/sparse?

cc @alexsamardzic @nikitaved @pearu @cpuhrsch @amjames @bhosmer

**Comment by alexsamardzic** [comment]
Assuming that the sparsity pattern you described is like:
```
t = torch.tensor([[0, 0, 0, 0, 0],
                  [0, 1, 0, 1, 0],
                  [0, 0, 0, 0, 0],
                  [0, 0, 0, 0, 0],
                  [1, 1, 0, 1, 1],
                  [0, 0, 0, 0, 0],
                  [0, 0, 0, 0, 0],
                  [0, 0, 0, 0, 0],
                  [1, 0, 1, 0, 0]])
```
then one could get what you call "fully-sparse" tensor by:
```
t.to_sparse()
```
and "hybrid sparse-dense" tensor by:
```
t.to_sparse(sparse_dims=1)
```

**Comment by nikitaved** [comment]
To add on the suggestion by @alexsamardzic , if it is the columns you are after, you would need to transpose your input beforehand. The downside is that the resulting tensor will be transposed, this is because the sparse dims have to be the left-most, unfortunately.

**Comment by ej159** [comment]
What I was trying to do can be accomplished by using bsc sparse tensors (with one redundant index array of all 0s). Following on from the example from @alexsamardzic:

```
u = t.to_sparse_csc()
v = u.to_sparse_bsc(blocksize=(u.size()[0],1))
```

I was just trying to extract dense columns from a sparse_csc tensor while not being able to go via a fully dense intermediate step and keeping everything within sparse tensors.
